tests/test-rest-posts-controller.php

```diff
@@ -357,6 +358,88 @@ public function test_get_items_tags_and_categories_query() {
 		$this->assertCount( 1, $response->get_data() );
 	}
 
+	public function test_get_items_sticky_query() {
+		$id1 = $this->post_id;
+		$id2 = $this->factory->post->create( array( 'post_status' => 'publish' ) );
+
+		update_option( 'sticky_posts', array( $id2 ) );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'sticky', true );
+
+		$response = $this->server->dispatch( $request );
+		$this->assertCount( 1, $response->get_data() );
+
+		$posts = $response->get_data();
+		$post = $posts[0];
+		$this->assertEquals( $id2, $post['id'] );
+	}
+
+	public function test_get_items_sticky_with_post__in_query() {
+		$id1 = $this->post_id;
+		$id2 = $this->factory->post->create( array( 'post_status' => 'publish' ) );
+		$id3 = $this->factory->post->create( array( 'post_status' => 'publish' ) );
+
+		update_option( 'sticky_posts', array( $id2 ) );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'sticky', true );
+		$request->set_param( 'include', array( $id1 ) );
+
+		$response = $this->server->dispatch( $request );
+		$this->assertCount( 0, $response->get_data() );
+
+		update_option( 'sticky_posts', array( $id1, $id2 ) );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'sticky', true );
+		$request->set_param( 'include', array( $id1 ) );
+
+		$response = $this->server->dispatch( $request );
+
+		$this->assertCount( 1, $response->get_data() );
+
+		$posts = $response->get_data();
+		$post = $posts[0];
+		$this->assertEquals( $id1, $post['id'] );
+	}
+
+	public function test_get_items_not_sticky_query() {
+		$id1 = $this->post_id;
+		$id2 = $this->factory->post->create( array( 'post_status' => 'publish' ) );
+
+		update_option( 'sticky_posts', array( $id2 ) );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'sticky', false );
+
+		$response = $this->server->dispatch( $request );
+		$this->assertCount( 1, $response->get_data() );
+
+		$posts = $response->get_data();
+		$post = $posts[0];
+		$this->assertEquals( $id1, $post['id'] );
+	}
```

Comment:
After discussion w/ @BE-Webdesign & @rmccue testing this case independently is probably overkill once the right sanitize callback is put in place